### PR TITLE
Stops the wheel if the arm is down

### PR DESCRIPTION
**Pull request by TheHarmonicRealm**

### CODE REVIEW

File: src/main/java/com/spartronics4915/frc2020/commands/PanelRotatorCommands.java [hunk 1]
```diff
@@ -1,5 +1,6 @@
 package com.spartronics4915.frc2020.commands;
 
+import com.fasterxml.jackson.databind.deser.std.MapEntryDeserializer;
```
Comment (omentic) [inline src/main/java/com/spartronics4915/frc2020/commands/PanelRotatorCommands.java]: ???

File: src/main/java/com/spartronics4915/frc2020/commands/PanelRotatorCommands.java [hunk 2]
```diff
@@ -40,6 +35,47 @@ public Raise(PanelRotator panelRotator)
                 panelRotator::getOpticalFlagUp, panelRotator);
         }
     }
+    */
+
+    public class Raise extends CommandBase
```
Comment (omentic) [inline src/main/java/com/spartronics4915/frc2020/commands/PanelRotatorCommands.java:43]: You can consolidate this all to a FunctionalCommand

File: src/main/java/com/spartronics4915/frc2020/commands/PanelRotatorCommands.java [hunk 3]
```diff
@@ -25,13 +26,7 @@
...
      */
 
-    /**
-     * This Raise {@link FunctionalCommand} calls {@link PanelRotator}.raise
-     * repeatedly, until the upper optical flag is broken, at which point the
-     * motor will stop.
-     * <p>
-     * The motor will also stop raising if interrupted by another Command.
-     */
+    /*
```
Comment (omentic) [inline src/main/java/com/spartronics4915/frc2020/commands/PanelRotatorCommands.java]: Add back in documentation

```
/**
 * This Raise {@link FunctionalCommand} calls {@link PanelRotator}.raise
 * repeatedly, until the upper optical flag is broken, at which point the
 * motor will stop.
 * <p>
 * The motor will also stop raising if interrupted by another Command.
 */
```

File: src/main/java/com/spartronics4915/frc2020/subsystems/PanelRotator.java [hunk 4]
```diff
@@ -16,22 +16,25 @@
...
-    private SpartronicsMotor mSpinMotor;
-    private SpartronicsMotor mRaiseMotor;
+    private final DigitalInput mProximitySensor;
     private final DigitalInput mOpticalFlagUp;
     private final DigitalInput mLimitSwitchDown;
-    private final ColorSensorV3 mColorSensor;
+    private SpartronicsMotor mSpinMotor;
+    private SpartronicsMotor mRaiseMotor;
 
+    private final ColorSensorV3 mColorSensor;
```
Comment (omentic) [inline src/main/java/com/spartronics4915/frc2020/subsystems/PanelRotator.java]: Move where these hardware are

File: src/main/java/com/spartronics4915/frc2020/subsystems/PanelRotator.java [hunk 5]
```diff
@@ -55,6 +58,17 @@ public PanelRotator()
         }
     }
 
+    /**
```
Comment (omentic) [inline src/main/java/com/spartronics4915/frc2020/subsystems/PanelRotator.java]: Proximity sensor has been cut.

File: src/main/java/com/spartronics4915/frc2020/subsystems/PanelRotator.java [hunk 6]
```diff
@@ -16,22 +16,25 @@
...
 
+    private final ColorSensorV3 mColorSensor;
     private String sensedColor;
     private String rotatedColor;
 
     private final ColorMatch mColorMatcher = new ColorMatch();
 
     public PanelRotator()
     {
+        mProximitySensor = new DigitalInput(Constants.PanelRotator.kProximitySensorId);
```
Comment (omentic) [inline src/main/java/com/spartronics4915/frc2020/subsystems/PanelRotator.java]: Proximity sensor has been cut

File: src/main/java/com/spartronics4915/frc2020/subsystems/PanelRotator.java [hunk 7]
```diff
@@ -149,12 +163,16 @@ else if (match.color.equals(Constants.PanelRotator.kYellowTarget))
...
-     * The position of our color sensor and the field's has a difference of π/2, so
-     * we need to adjust targets accordingly.
-     * <p>
-     * See https://drive.google.com/uc?id=1BfoFJmpJg31txUqTG-OrJjeWgQdQsCNC
-     * for a diagram of how these line up.
-     * <p>
+     * Finds the distance between the sensor and what it is looking at
+     * @return 11-bit (0-2047) value (to dashboard)
+     */
+    public void getDistance()
```
Comment (omentic) [inline src/main/java/com/spartronics4915/frc2020/subsystems/PanelRotator.java]: Ooh, maybe using this would be more reliable than `getColorConfidence()`.

Comment (omentic) [reply src/main/java/com/spartronics4915/frc2020/subsystems/PanelRotator.java]: We should test that if possible.

File: src/main/java/com/spartronics4915/frc2020/subsystems/PanelRotator.java [hunk 8]
```diff
@@ -93,33 +107,33 @@ public String getTargetColor()
...
+    public void get18BitRGB()
     {
         int red = mColorSensor.getRed();
         int green = mColorSensor.getGreen();
         int blue = mColorSensor.getBlue();
 
         String RGB = red + ", " + green + ", " + blue;
 
-        return RGB;
+        this.dashboardPutString("Color sensor 18 Bit", RGB);
```
Comment (omentic) [inline src/main/java/com/spartronics4915/frc2020/subsystems/PanelRotator.java]: What's the idea behind this change?

**Review by omentic — CHANGES_REQUESTED** [review]

File: src/main/java/com/spartronics4915/frc2020/commands/PanelRotatorCommands.java (hunk 1 above)
Comment (TheHarmonicRealm) [reply src/main/java/com/spartronics4915/frc2020/commands/PanelRotatorCommands.java]: I have no idea where this came from

**Comment by omentic** [comment]
There's some really weird issues with this PR, GitHub / git seems to think changes that haven't been made have been (mostly around subsystem pass-through). @TheHarmonicRealm I'm going to `git cherry-pick` your commits into `master`...

**Pull request closed without merging**